File: by_hand/lib/models/product.dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

class Products{
  int id;
  int userId;
  int categoryId;
  bool available;
  bool favorite;
  String image1;
  String image2;
  String name;
  String subName;
  String price;
  String description;
  String type;
  String rates;


  Products();

  Products.fromJson(Map<String,dynamic> json){
    this.id=json['id'];
    this.userId=json['user_id'];
    this.categoryId=json['category_id'];
    this.available=json['available'];
    this.image1=json['image1_url'];
    this.image2=json['image2_url'];
    this.name=json['name'];
    this.subName=json['sub_name'];
    this.price=json['price'];
    this.description=json['description'];
    this.type=json['type'];
    this.rates=json['rates'];

  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'user_id': userId,
    'category_id': categoryId,
    'available': available,
    'image1_url': image1,
    'image2_url': image2,
    'name': name,
    'sub_name': subName,
    'price': price,
    'description': description,
    'type': type,

  };
}

File: by_hand_api/src/models/product.js
import { Model } from 'sequelize';


export default (sequelize, DataTypes) => {
  class Products extends Model {
   
     
    static associate(models) {
      // define association here
      Products.belongsTo(models.User,{foreignKey:"user_id"})
      Products.belongsTo(models.Categorys,{foreignKey:"category_id"})


      Products.hasMany(models.Rates,{foreignKey:"product_id"})
      Products.hasOne(models.Favorites,{foreignKey:"product_id"})



    }
  };
  Products.init({
    user_id: DataTypes.INTEGER,
    category_id: DataTypes.INTEGER,
    available: DataTypes.BOOLEAN,
    image1_url:DataTypes.STRING,
    image2_url:DataTypes.STRING,
    name: DataTypes.STRING,
    sub_name: DataTypes.STRING,
    price: DataTypes.STRING,
    description: DataTypes.STRING,
    deletedAt: 'destroyTime'

  }, {
    paranoid: true,
    sequelize,
    tableName:'product',
    modelName: 'Products',
  });
  return Products;
};